llvm/lib/CodeGen/RegisterCoalescer.cpp

```diff
@@ -1339,6 +1334,32 @@ bool RegisterCoalescer::reMaterializeTrivialDef(const CoalescerPair &CP,
   if (SrcIdx && DstIdx)
     return false;
 
+  // Only support subregister destinations when the def is read-undef.
+  MachineOperand &DstOperand = CopyMI->getOperand(0);
+  Register CopyDstReg = DstOperand.getReg();
+  if (DstOperand.getSubReg() && !DstOperand.isUndef())
+    return false;
+
+  // Only support subregister destinations when the def is read-undef, in the
+  // physical register case. We're widening the def and need to avoid clobbering
```

Comment:
I would start the comment with: "In the physical register case checking that the def is read-undef is not enough [...]" instead of repeating the beginning of the previous comment.